File: 0001-0500/0062.Unique Paths.rb
# @param {Integer} m
# @param {Integer} n
# @return {Integer}
def unique_paths(m, n)
    dp = Array.new(n, 1)
    for _ in (1...m)
        for j in (1...n)
            dp[j] += dp[j - 1]
        end
    end
    dp[-1]
end

File: 0001-0500/0062.Unique Paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        if m > n:
            return self.uniquePaths(n, m)
        dp = [1] * m
        for _ in xrange(n - 1):
            for j in xrange(1, m):
                dp[j] += dp[j - 1]
        return dp[-1]
